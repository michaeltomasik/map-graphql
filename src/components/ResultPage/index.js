import React, { Component, Fragment } from 'react'
import { Query } from 'react-apollo'
import  { gql } from 'apollo-boost'

import GraphQLResultAwait from '../GraphQLResultAwait'

export default class ResultPage extends Component {
  render() {
    return (
      <Query query={ARTICLES_QUERY} variables={{ matchingString: this.props.match.params.matchingString }}>
        {({ data, loading, error, refetch }) => {
          if (loading || error) {
            return <GraphQLResultAwait  loading={loading} error={error} />
          }

          return (
            <Fragment>
              <h1>Results:</h1>
              <ul>{
                data.getArticles.items.map(article =>
                  <li key={article.link}>
                    <a href={article.link}>{article.title}</a>
                  </li>)
              }</ul>
            </Fragment>
          )
        }}
      </Query>
    )
  }
}

export const ARTICLES_QUERY = gql`
  query Articles($matchingString: String) {
    getArticles(matchingString: $matchingString){
        items {
          title
          link
        }
    }
  }
`