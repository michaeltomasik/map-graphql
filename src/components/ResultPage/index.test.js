import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils';
import { mount } from 'enzyme';
// The component AND the query need to be exported
import ResultPage, { ARTICLES_QUERY } from './index';
import wait from 'waait';

const mocks = [
  {
    request: {
      query: ARTICLES_QUERY,
      variables: {
        matchingString: 'UK',
      },
    },
    result: {
      data: {
        getArticles: {
          items: [
            {
              "link": "www.Nepal.com",
              "title": "NP",
            },
            {
              "link": "www.test.com",
              "title": "test",
            },
          ]
        },
      },
    },
  },
];

const props = {
  match: {
    params: {
      matchingString: 'UK',
    }
  }
}

describe('ResultPage form test suite', () => {
  it('renders with data', async () => {
    const wrapper = mount((
      <MockedProvider mocks={mocks} addTypename={false}>
        <ResultPage {...props} />
      </MockedProvider>
    ));
    // https://github.com/apollographql/react-apollo/issues/2719
    await wait(0);
    wrapper.update()

    expect(wrapper.debug()).toMatchSnapshot();
  });
});