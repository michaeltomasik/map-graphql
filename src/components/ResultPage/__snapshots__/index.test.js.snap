// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPage form test suite renders with data 1`] = `
"<MockedProvider mocks={{...}} addTypename={false}>
  <ApolloProvider client={{...}}>
    <ResultPage match={{...}}>
      <Query query={{...}} variables={{...}}>
        <h1>
          Results:
        </h1>
        <ul>
          <li>
            <a href=\\"www.Nepal.com\\">
              NP
            </a>
          </li>
          <li>
            <a href=\\"www.test.com\\">
              test
            </a>
          </li>
        </ul>
      </Query>
    </ResultPage>
  </ApolloProvider>
</MockedProvider>"
`;
